astropy/utils/iers/iers.py

```diff
@@ -679,20 +711,28 @@ def open(cls):
             if cls.iers_table.meta.get('data_url') in all_urls:
                 return cls.iers_table
 
-        try:
-            filename = download_file(all_urls[0], sources=all_urls, cache=True)
-        except Exception as err:
+        for url in all_urls:
+            try:
+                filename = download_file(url, cache=True)
+            except Exception as err:
+                warn(f'failed to download {url}: {err}', IERSWarning)
```

Comment:
```suggestion
                warn(f'failed to download {url}: {repr(err)}', IERSWarning)
```

Reply:
@pllim - are you sure about wanting `repr()` here? I don't think that is common for exception messages since it turns newlines into the `\n` character and can otherwise make the output messy.

Reply:
Oh, I thought `repr(err)` was recommended in one of the PyCon talks a few years back. If you prefer not to use it, that is fine too.

Reply:
If you have a reference I'd be interested to look but for now let's leave it.

Reply:
I don't feel like watching the talk right now but I remember ejeschke/ginga#767 , FWIW.

Reply:
I watched it and this is an interesting question:
```
In [3]: try:
   ...:     raise ValueError('blah\nblah\nblah\n')
   ...: except Exception as exc:
   ...:     print(f'failed to download: {repr(exc)}')
   ...: 
failed to download: ValueError('blah\nblah\nblah\n')

In [4]: try:
   ...:     raise ValueError('blah\nblah\nblah\n')
   ...: except Exception as exc:
   ...:     print(f'failed to download: {exc}')
   ...: 
failed to download: blah
blah
blah
```
I'm not sure about this, but I am definitely sure that I am tired of this PR.. :smile:

Reply:
Let's leave it then. 😉

Reply:
I'd actually add `!r` for the `repr`  (i.e., `{exc!r}`) - it is useful to know what the type of the exception was!